src/libpmemfile-posix/data.c

```diff
@@ -568,8 +571,13 @@ file_write(PMEMfilepool *pfp, PMEMfile *file, struct pmemfile_inode *inode,
 
 	/* All blocks needed for writing are properly allocated at this point */
 
-	iterate_on_file_range(pfp, file, file->offset, count, (char *)buf,
-	    write_to_blocks);
+	struct pmemfile_block *block = file_find_block(file, file->offset);
+
+	block = iterate_on_file_range(pfp, file->vinode, block, file->offset,
+			count, (char *)buf, write_to_blocks);
+
+	if (block)
+		file->block_pointer_cache = block;
```

Comment:
This sets the `block_pointer_cache` to the last block of the range, doesn't it?
I think, to be more useful for the next `iterate_on_file_range` call, the code should rather remember the first block of the range.